### ISSUE TITLE
Improve context of utility button for screen reader users

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#279]
# Feature Request

## Description
When navigating to https://engineering.cerner.com/terra-ui/#/components/terra-application-utility/application-utility/application-utility and testing VoiceOver, I noticed the utility button is always read as "Utility Button". I think it would be better if we encourage users to pass in meaningful text for this that gets communicated to screen readers.

We could either set whatever string they pass to the title prop as the aria-label or add a new prop specifically to control the aria-label along with some doc on why one would want to do this.

**Comment by gabeparra01** [comment]
Just to clarify, the code changes for this issue will be added to terra-button? Or other places too?

**Comment by bjankord** [comment]
terra-button supports aria-label so no code changes should be needed there. The issue this is for is that the Utility Button in the examples visually displays the string "User Name", but to screen readers, the button is announced as "Utility Button". Need some guidance from UX on how this button should be announced. My thought it we should probably open up a prop in terra-application-utility that lets app teams control the aria-label on that button.

**Comment by bjankord** [comment]
Resolved in https://github.com/cerner/terra-framework/pull/834
